src/tribler/gui/core_manager.py

```diff
@@ -143,10 +143,12 @@ def stop(self, quit_app_on_core_finished=True):
 
         self.shutting_down = True
         self._logger.info("Stopping Core manager")
-        if self.core_process or self.core_connected:
+        if (self.core_process or self.core_connected) and not self.core_finished:
```

Comment:
NIT: for me it is more clear
```suggestion
        need_to_shutdown_core = (self.core_process or self.core_connected) and not self.core_finished
        if need_to_shutdown_core:
```

Reply:
Thanks, changed the condition this way